### PR TITLE
GVRAssetLoader import vertex colors

### PR DESCRIPTION
**Pull request by NolaDonato**
This patch adds support for importing vertex colors from assimp. It does not affect X3d models.
Addresses enhancement request #1077

GearVRF DCO signed off by: Nola Donato nola.donato@samsung.com

### CODE REVIEW

**Comment by mwitchwilliams** [comment]
There is an X3D Color node that GearVR's X3D implementation does not currently support.  The Color node includes a boolean colorPerVertex.  With this added support in GearVRf, it seems we could now support X3D's Color node when colorPerVertex is true.  When colorPerVertex is false, X3D expects each color for a polygon, which there is no plan to support in GearVR.

**Comment by NolaDonato** [comment]
Great! Try to add the support in X3D and import some models with vertex colors. If that works, we can add X3D vertex color support to the pull request to make it complete.

File: GVRf/Framework/framework/src/main/java/org/gearvrf/GVRJassimpAdapter.java [hunk 1]
```diff
@@ -134,6 +137,22 @@ public GVRMesh createMesh(GVRContext ctx, AiMesh aiMesh) {
             }
         }
 
+        // Vertex Colors
+        for(int c = 0; c < MAX_VERTEX_COLORS; c++) {
+            FloatBuffer fbuf = aiMesh.getColorBuffer(c);
+            if (fbuf != null) {
+                FloatBuffer coords = FloatBuffer.allocate(aiMesh.getNumVertices() * 4);
+                FloatBuffer source = aiMesh.getColorBuffer(c);
+                String name = "a_color";
```
Comment (thomasflynn) [inline GVRf/Framework/framework/src/main/java/org/gearvrf/GVRJassimpAdapter.java]: quick question on this.  if the model does have multiple 'vertex color' arrays (for whatever reason), it looks like they'll all get the same name.  that would be a problem, wouldn't it?

Comment (NolaDonato) [inline GVRf/Framework/framework/src/main/java/org/gearvrf/GVRJassimpAdapter.java]: The full source looks like this (c is the color set index):
                String name = "a_color";

                if (c > 0) {
                    name += c;
                }

Comment (thomasflynn) [inline GVRf/Framework/framework/src/main/java/org/gearvrf/GVRJassimpAdapter.java]: ah, somehow i missed that.  Ok, thanks.  Code looks good then.

**Comment by thomasflynn** [comment]
ok, with @NolaDonato 's help, I was able to run the asset test and verify this loads the model as expected.  +1 Verified.  +1 Looks good to me.

**Pull request merged**